### PR TITLE
[react] Deprecate ReactChildren

### PR DESCRIPTION
**Pull request by eps1lon**
The type can be (and [is currently](https://twitter.com/HeikoMathes/status/1513399571753230341)) confused with something that you assign to `children`.
So instead, we just declare the type directly on `React.Children`.

### CODE REVIEW

File: types/moxy__next-router-scroll/RouterScrollProvider.d.ts [hunk 1]
```diff
@@ -20,6 +20,6 @@ declare namespace ScrollBehaviorProvider {
     namespace propTypes {
         const disableNextLinkScroll: boolean;
         const shouldUpdateScroll: any;
-        const children: ReactChildren;
+        const children: ReactNode;
```
Comment (eps1lon) [inline types/moxy__next-router-scroll/RouterScrollProvider.d.ts:23]: https://github.com/moxystudio/next-router-scroll/blob/v2.2.0/src/RouterScrollProvider.js#L78

File: types/react-mentions/lib/utils/readConfigFromChildren.d.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { ReactChildren } from 'react';
+import { ReactNode } from 'react';
 import { Config } from './applyChangeToValue';
 
-export function readConfigFromChildren(children: ReactChildren): Config[];
+export function readConfigFromChildren(children: ReactNode): Config[];
```
Comment (eps1lon) [inline types/react-mentions/lib/utils/readConfigFromChildren.d.ts:4]: https://github.com/signavio/react-mentions/blob/v4.3.1/src/utils/readConfigFromChildren.js#L6-L7

File: types/react-native-table-component/index.d.ts [hunk 3]
```diff
@@ -76,7 +76,7 @@ export interface TableProps {
 }
 
 export class Table extends Component<TableProps> {
-  _renderChildren(props: TableProps): ReactChildren;
+  _renderChildren(props: TableProps): ReactNode;
```
Comment (eps1lon) [inline types/react-native-table-component/index.d.ts:79]: https://github.com/Gil2015/react-native-table-component/blob/b2ff845e91a48db135756e2bebe36b157bdafaeb/components/table.js#L11

File: types/react-native-table-component/index.d.ts [hunk 4]
```diff
@@ -86,5 +86,5 @@ export interface TableWrapperProps {
 }
 
 export class TableWrapper extends Component<TableWrapperProps> {
-  _renderChildren(props: TableWrapperProps): ReactChildren;
+  _renderChildren(props: TableWrapperProps): ReactNode;
```
Comment (eps1lon) [inline types/react-native-table-component/index.d.ts:89]: https://github.com/Gil2015/react-native-table-component/blob/b2ff845e91a48db135756e2bebe36b157bdafaeb/components/table.js#L49-L51

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react@latest/))
was missing the following properties:
1. unstable_act
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7788ae539846ade5c29ee9a188bc1342ff044f2d
</p>

**Review by johnnyreilly — APPROVED** [review]

**Comment by johnnyreilly** [comment]
Looks good

**Comment by eps1lon** [comment]
Ready to merge

**Pull request merged**